bluesky/protocols.py

```diff
@@ -244,18 +325,32 @@ def check_value(self, value: Any) -> Union[NoReturn, None]:
 
         This method is used by simulators that check limits. If not implemented
         those simulators should assume all values are valid, but may warn.
+
+        This method may be used during a scan, so should not write to any Signals
         """
         ...
 
 
+class Hints(TypedDict, total=False):
+    fields: List[str]
+    dimensions: List[Tuple[List[str], Literal["primary"]]]
+    gridding: Literal["rectilinear", "rectilinear_nonsequential"]
+
+
 @runtime_checkable
-class Hinted(Protocol):
+class HasHints(HasName, Protocol):
```

Comment:
Following @danielballan's suggestion of `Has*`, we now have `HasName`, `HasParent` and `HasHints`